--- frameworks/base/core/java/com/google/android/mms/pdu/PduPersister.java	2013-08-27 20:31:41.460931384 -0600
+++ patches/frameworks/base/core/java/com/google/android/mms/pdu/PduPersister.java	2013-08-27 19:50:06.840906087 -0600
@@ -1341,6 +1341,12 @@
                 String phoneNo = mTelephonyMgr.getLine1Number();
                 array = addressMap.get(PduHeaders.TO);
 
+				// Temporary fix for NullPointerException - var: phoneNo
+				// Applied by titanic-fanatic
+				if (null == phoneNo){
+					phoneNo = "";
+				}
+
                 if (array != null) {
                     for (EncodedStringValue v : array) {
                         if (v != null && !v.getString().replaceAll("\\D", "").equals(phoneNo.replaceAll("\\D", ""))) {
@@ -1499,3 +1505,4 @@
                 PendingMessages.DUE_TIME);
     }
 }
+
